src/commands/deploy.ts

```diff
@@ -66,6 +67,21 @@ export async function deploy(context: IActionContext, confirmDeployment: boolean
         }
     }
 
+    let siteConfig: WebSiteModels.SiteConfigResource | undefined;
+
+    if (!fsPath) {
+        // we can only get the siteConfig if the entry point was a treeItem
+        siteConfig = node ? await node.root.client.getSiteConfig() : undefined;
+
+        if (siteConfig && javaUtils.isJavaRuntime(siteConfig.linuxFxVersion)) {
+            const fileExtension: string = javaUtils.getArtifactTypeByJavaRuntime(siteConfig.linuxFxVersion);
+            fsPath = await javaUtils.showQuickPickByFileExtension(context, `Select the ${fileExtension} file to deploy...`, fileExtension);
```

Comment:
Can't you get rid of `javaUtils.showQuickPickByFileExtension` and just use `showWorkspaceFoldersAndFiles` in all cases? I don't like having different code-paths/prompts for command palette vs. right click on a node.